src/WCSimRunAction.cc

```diff
@@ -514,9 +527,10 @@ void WCSimRunAction::EndOfRunAction(const G4Run*)
     // Close the Root file at the end of the run
     TFile* hfile = WCSimTree->GetCurrentFile();
     hfile->cd();
-    optionsTree->Fill();
-    optionsTree->Write();
-    
+    if(aRun->GetRunID() == 0) {
```

Comment:
Is it possible to make this be filled on every run? Things don't change in your example, but if someone e.g. changed the dark rate, it would be useful

Hopefully ROOT's compression algorithm is sensible and the overhead for having multiple identical entries in a tree is small

Reply:
Yes, can do this. Initially I decided against doing so because it would not be clear which options tree entries corresponded to which events. But on further investigation I found that the WCSimRootEvent has an EventHeader with a run number. It's currently hard coded to be always 0, but I'll change it to correspond to the actual run number and therefore correspond to the relevent options tree entry.

Reply:
That's even better. Thanks Nick!